### ISSUE TITLE
 UnhandledPromiseRejectionWarning: TypeError Received an instance of SourceMapGenerator

### ISSUE THREAD

**Issue by steinmb** [postcss/postcss#1370]
Hi
Not 100% it belong here. I am no js-devloper so please be gentle.

```
 node --version
v14.4.0

./node_modules/postcss-cli/bin/postcss --version

                                      /|\
                                    //   //
                                  //       //
                                //___*___*___//
                              //--*---------*--//
                            /|| *             * ||/
                          // ||*    v6.1.3     *|| //
                        //   || *             * ||   //
                      //_____||___*_________*___||_____//

```
Running tests with online css-file. containing only `h1 {margin: 1em;}` I generate the out.css file but as you can see. A lot of warnings.

```
node --trace-warnings ./node_modules/postcss-cli/ test.css -o out.css --no-map
(node:64921) UnhandledPromiseRejectionWarning: TypeError [ERR_INVALID_ARG_TYPE]: The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received an instance of SourceMapGenerator
    at writeFile (fs.js:1356:5)
    at go$writeFile (/Users/steinmb/sites/olivero/node_modules/graceful-fs/graceful-fs.js:131:14)
    at Object.writeFile (/Users/steinmb/sites/olivero/node_modules/graceful-fs/graceful-fs.js:128:12)
    at /Users/steinmb/sites/olivero/node_modules/fs-extra/lib/output/index.js:18:27
    at /Users/steinmb/sites/olivero/node_modules/universalify/index.js:23:46
    at emitUnhandledRejectionWarning (internal/process/promises.js:149:15)
    at processPromiseRejections (internal/process/promises.js:211:11)
    at processTicksAndRejections (internal/process/task_queues.js:98:32)
(node:64921) TypeError [ERR_INVALID_ARG_TYPE]: The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received an instance of SourceMapGenerator
    at writeFile (fs.js:1356:5)
    at go$writeFile (/Users/steinmb/sites/olivero/node_modules/graceful-fs/graceful-fs.js:131:14)
    at Object.writeFile (/Users/steinmb/sites/olivero/node_modules/graceful-fs/graceful-fs.js:128:12)
    at /Users/steinmb/sites/olivero/node_modules/fs-extra/lib/output/index.js:18:27
    at /Users/steinmb/sites/olivero/node_modules/universalify/index.js:23:46
(node:64921) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
    at emitDeprecationWarning (internal/process/promises.js:161:11)
    at processPromiseRejections (internal/process/promises.js:213:13)
    at processTicksAndRejections (internal/process/task_queues.js:98:32)
```

**Comment by steinmb** [comment]
That was just the postcss-cli version. Here is full dependency tree. It makes me think perhaps the versions it include is outdated and that might be the reason it throws all the warnings? 

* [packages.json.txt](https://github.com/postcss/postcss/files/4798551/packages.json.txt)
* [yarn.lock.txt](https://github.com/postcss/postcss/files/4798528/yarn.lock.txt)

I'm a 'little' out of my element here but trying my best :)

**Comment by ai** [comment]
Sorry. This issue is related to `postcss-cli`. You should report it there.

**Comment by eemeli** [comment]
I hit this issue with postcss 7.0.34, i.e. not with postcss-cli. The root issue is the same as solved by postcss/postcss-cli#324: `await postcss.process(...).map` used to be a string, but now it's a SourceMapGenerator, and needs an additional `String()` wrapper when used in a context that doesn't automatically call its `toString()` method.

I'm pretty sure this changed sometime after 7.0.17, but I can't be absolutely certain.
